yaqd-core/yaqd_core/_has_transformed_position.py

```diff
@@ -0,0 +1,88 @@
+__all__ = ["HasTransformedPosition"]
+
+
+import pathlib
+from typing import Dict, Any, Optional, List
+from abc import abstractmethod
+
+from yaqd_core import HasLimits, HasPosition, IsDaemon
+
+
+class HasTransformedPosition(HasLimits, HasPosition, IsDaemon):
+    def __init__(
+        self, name: str, config: Dict[str, Any], config_filepath: pathlib.Path
+    ):
+        super().__init__(name, config, config_filepath)
+
+    # --- conversion of coordinates ---------------------------------------------------------------
+
+    def to_native(self, transformed_position):
+        relative = self._transformed_to_relative(transformed_position)
+        return relative + self._state["native_reference_position"]
+
+    def to_transformed(self, native_position):
+        relative = native_position - self._state["native_reference_position"]
+        return self._relative_to_transformed(self, relative)
+
+    @abstractmethod
+    def _relative_to_transformed(self, relative_position):
+        """convert a relative coordinate to a transformed coordinate.
+        Relative coordinates differ from natural coordinates in that the null position has been subtracted.
+        (i.e. in relative coordinates reference position is zero).
+        The inverse function of `_transformed_to_relative`.
+
+        If you needo only an offset, return relative_position
```

Comment:
I wonder if we should make this behavior default, its a reasonably sane default and lowers barrier of entry for the simple case.


Thoughts @untzag @ddkohler ?

Reply:
Yeah, I can get on board with default being it just works with `reference_position` offsets.